kotlinx-coroutines-core/jvm/src/internal/ResizableAtomicArray.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2016-2022 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.internal
+
+import java.util.concurrent.atomic.*
+
+/**
+ * Atomic array with lock-free reads and synchronized modifications. It logically has an unbounded size,
+ * is implicitly filled with nulls, and is resized on updates as needed to grow.
+ */
+internal class ResizableAtomicArray<T>(initialLength: Int) {
```

Comment:
Let's add a Lincheck test

Reply:
Ok. Added. Though it still cannot catch, for example, a missing `@Volatile` annotation in this particular code.